kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -77,9 +81,12 @@ class ActionsIterator implements CloseableIterator<ActionWrapper> {
         this.tableClient = tableClient;
         this.checkpointPredicate = checkpointPredicate;
         this.filesList = new LinkedList<>();
-        this.filesList.addAll(files);
+        this.filesList.addAll(
+                files.stream().map(file -> DeltaLogFile.forCommitOrCheckpoint(file))
+                        .collect(Collectors.toList()));
         this.readSchema = readSchema;
         this.actionsIter = Optional.empty();
+        this.schemaContainsAddOrRemoveFiles = LogReplay.containsAddOrRemoveFileActions(readSchema);
```

Comment:
When is this false? Won't the json/checkpoint schema always have add or remove files?

Reply:
This is false when we are reading only the metadata and protocol. In this case, we don't want to handle sidecars at all (because we don't have to).